src/main/kotlin/org/rust/lang/core/psi/impl/mixin/RsImplItemImplMixin.kt

```diff
@@ -31,3 +30,19 @@ abstract class RsImplItemImplMixin : RsStubbedElementImpl<RsImplItemStub>, RsImp
         return PresentationData(typeReference?.text ?: "Impl", null, RsIcons.IMPL, null)
     }
 }
+
+/**
+ * @return pair of two lists: (mandatory trait members, optional trait members)
+ */
+fun RsImplItem.toImplementOverride(): Pair<List<RsNamedElement>, List<RsNamedElement>> {
+    val trait = traitRef?.trait ?: error("No trait ref")
```

Comment:
Here should be `?: return null` instead of throwing error. Resolve engine is not perfect now, so we don't guarantee resolving all the possible trait references (let alone the cases when the a trait name is typed incorrectly). In such cases it's better to tell the caller that we can't collect items and just don't show the window instead of producing an IDE error.